### PR TITLE
Add support for dashboards API

### PR DESCRIPTION
**Pull request by smithclay**
Depends on: https://github.com/paultyng/go-newrelic/pull/14

Adds support for the new Dashboard API, this will let people write and manage dashboards as code.

still needs:
- [x] [acceptance test](https://github.com/hashicorp/terraform/blob/master/.github/CONTRIBUTING.md#writing-acceptance-tests) coverage
- [x] add website docs

Docs: https://docs.newrelic.com/docs/insights/insights-api/manage-dashboards/insights-dashboard-api
Announcement: https://blog.newrelic.com/2018/01/08/insights-dashboard-api/

### CODE REVIEW

**Comment by paultyng** [comment]
SDK pr merged and release cut

File: newrelic/resource_newrelic_dashboard.go [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+		Update: resourceNewRelicDashboardUpdate,
+		Delete: resourceNewRelicDashboardDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"title": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false,
```
Comment (paultyng) [inline newrelic/resource_newrelic_dashboard.go]: this defaults to `false`, so unnecessary

File: newrelic/resource_newrelic_dashboard.go [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/schema"
+	newrelic "github.com/paultyng/go-newrelic/api"
+)
+
+func stringInList(values []string) schema.SchemaValidateFunc {
```
Comment (paultyng) [inline newrelic/resource_newrelic_dashboard.go]: You can use this method instead: https://github.com/hashicorp/terraform/blob/master/helper/validation/validation.go#L74

File: newrelic/resource_newrelic_dashboard.go [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+
+			widgetLayout := newrelic.DashboardWidgetLayout{
+				Row:    w["row"].(int),
+				Column: w["column"].(int),
+				Width:  w["width"].(int),
+				Height: w["height"].(int),
+			}
+
+			// TODO: Support non-NRQL Widgets
+			widgetData := make([]newrelic.DashboardWidgetData, 0, 1)
```
Comment (paultyng) [inline newrelic/resource_newrelic_dashboard.go]: can just do this in a single line as a slice literal:

```
widgetData := []nr.DWD{
  {
    NRQL: ...,
  },
}
```

**Comment by paigebernier** [comment]
OK looks great! Let's bring dashboards to the people 💃

File: newrelic/resource_newrelic_dashboard_test.go [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+	}
+}
+
+func testAccCheckNewRelicDashboardWidgetConfigAdded(rName string) string {
+	return fmt.Sprintf(`
+resource "newrelic_dashboard" "foo" {
+  title                = "tf-test-updated-%s"
+  widget {
+    title         = "Average Transaction Duration"
+	visualization = "faceted_line_chart"
```
Comment (paultyng) [inline newrelic/resource_newrelic_dashboard_test.go]: got some weird whitespace stuff going on here

**Comment by smithclay** [comment]
Several changes today based on feedback from @radeksimko (thanks!)

**Comment by paultyng** [comment]
lgtm, needs rebase though

**Comment by paultyng** [comment]
:tada:

**Pull request merged**